impress-org/givewp
# Feature: New Donation and Donor list admin screens

Resolves #6328, #6305

## Description


The Donations page has needed a refresh, and this PR aims to do just that. This PR adds a new Donations admin page, as well as bulk actions for the Forms, Donors, and Donations pages. 

Some other changes:

- Added an option to filter by donor name/ID on the Donations page
- Deleting items using row actions now requires a confirmation step
- Added a button to switch between the old admin archive pages and the new ones - selection persisted in usermeta
- Added preloading to the Donation Forms archive page
- Some accessibility improvements to the updated Donation Forms table (which carry over into Donors and Donations)

## Affects


This PR does affect the legacy Donors, Donations, and Donation Forms archive pages. You can test those by going to an updated archive page and clicking the "Switch to Legacy View" button.

## Visuals



https://user-images.githubusercontent.com/7819986/163074264-8b3f989f-3174-44e1-8504-cec1c14763ed.mp4

## Testing Instructions



1. Install GiveWP
2. In the admin area, click on Donations -> Donations
3. Make sure the Donations list page appears
4. Check that the row actions (Edit, Delete) work correctly
5. Make sure that searching/filtering works correctly
6. For each of the archive pages (All Forms, Donors, Donations), try each of the bulk actions, making sure they work
7. For each of the archive pages (All Forms, Donors, Donations), click the button in the upper-right hand corner to switch to the legacy view
8. Make sure that each of the legacy views displays correctly
9. Make sure that switching from the legacy view to the updated view works correctly

## Pre-review Checklist
